runtime/src/accounts_db.rs

```diff
@@ -937,6 +937,26 @@ struct RemoveUnrootedSlotsSynchronization {
 
 type AccountInfoAccountsIndex = AccountsIndex<AccountInfo>;
 
+pub trait AccountsUpdateNotifierIntf: std::fmt::Debug {
+    /// Notified when an account is updated at runtime, due to transaction activities
+    fn notify_account_update(&self, slot: Slot, pubkey: &Pubkey, account: &AccountSharedData);
+
+    /// Notified when the AccountsDb is initialized at start when restored
+    /// from a snapshot.
+    fn notify_account_data_at_start(&self, slot: Slot, account: &StoredAccountMeta);
+
+    /// Notified when a slot is optimistically confirmed
+    fn notify_slot_confirmed(&self, slot: Slot, parent: Option<Slot>);
+
+    /// Notified when a slot is marked frozen.
+    fn notify_slot_processed(&self, slot: Slot, parent: Option<Slot>);
+
+    /// Notified when a slot is rooted.
+    fn notify_slot_rooted(&self, slot: Slot, parent: Option<Slot>);
+}
+
+pub type AccountsUpdateNotifier = Arc<RwLock<dyn AccountsUpdateNotifierIntf + Sync + Send>>;
+
```

Comment:
I think this might make sense to move to a separate module

Reply:
okay

Reply:
Done